src/core/thread/mle_types.hpp

```diff
@@ -567,6 +568,61 @@ class LeaderData : public otLeaderData
     void SetLeaderRouterId(uint8_t aRouterId) { mLeaderRouterId = aRouterId; }
 };
 
+OT_TOOL_PACKED_BEGIN
+class RouterIdSet
+{
+public:
+    /**
+     * This method clears the Router Id Set.
+     *
+     */
+    void Clear(void) { memset(mRouterIdSet, 0, sizeof(mRouterIdSet)); }
+
+    /**
+     * This method indicates whether or not a Router ID bit is set.
+     *
+     * @param[in]  aRouterId  The Router ID.
+     *
+     * @retval TRUE   If the Router ID bit is set.
+     * @retval FALSE  If the Router ID bit is not set.
+     *
+     */
+    bool Contains(uint8_t aRouterId) const { return (mRouterIdSet[aRouterId / 8] & (0x80 >> (aRouterId % 8))) != 0; }
+
+    /**
+     * This method sets a given Router ID.
+     *
+     * @param[in]  aRouterId  The Router ID to set.
+     *
+     */
+    void Add(uint8_t aRouterId) { mRouterIdSet[aRouterId / 8] |= 0x80 >> (aRouterId % 8); }
+
+    /**
+     * This method removes a given Router ID.
+     *
+     * @param[in]  aRouterId  The Router ID to remove.
+     *
+     */
+    void Remove(uint8_t aRouterId) { mRouterIdSet[aRouterId / 8] &= ~(0x80 >> (aRouterId % 8)); }
+
+    /**
+     * This method returns whether or not the Router ID sets are equal.
+     *
+     * @param[in]  aOther The other Router ID Set to compare with.
+     *
+     * @retval TRUE   If the Router ID sets are equal.
+     * @retval FALSE  If the Router ID sets are not equal.
+     *
+     */
+    bool operator!=(const RouterIdSet &aOther) const
+    {
+        return memcmp(mRouterIdSet, aOther.mRouterIdSet, sizeof(mRouterIdSet)) != 0;
+    }
```

Comment:
How about if we also add `operator==` as well and implement `!=` from it instead?

Reply:
updated as your suggestion.